macros/parsers/parserPopUp.pl

```diff
@@ -225,6 +251,30 @@ sub MENU {
 sub named_ans_rule           { shift->MENU(0, @_) }
 sub named_ans_rule_extension { shift->MENU(1, @_) }
 
+#
+# DropDown() variant of PopUp() with placeholder
+#
+
+sub DropDown {
+	my ($self, $choices, $value, %options) = @_;
+	$options{placeholder} = '?' unless $options{placeholder};
+	unshift(@$choices, $options{placeholder});
+	my @order = map { ref($_) eq "ARRAY" ? @$_ : $_ } @$choices;
+	if ($value =~ m/^\d+$/ && $order[$value]) { $value++ }
+	return parser::PopUp->new($choices, $value, \%options);
+}
+
+#
+# TrueFalse() variant of PopUp()
+#
+
+sub DropDownTF {
+	my ($self, $value, %options) = @_;
+	my $sanitized_value = $value ? 'True' : 'False';
+	$sanitized_value = 'False' if (defined $value && $value =~ /^\s*(false|f)\s*$/i);
```

Comment:
If you localize this, the regular expression gets harder.  If one assumes that true and false have different first letters in all languages, then if you have the localized "True" in `$true` and the localized "False" in `$false`, then you might be able to do something like

``` perl
sub DropDownTF {
	my $self    = shift;
	my $value   = shift;   ### you will need to call the localization on this value to get the localized T/F
	my %options = (
	  static => 0,
	  @_
	);
	my $sanitized_value = ({
	  lc(substr($true, 0, 1)) => $true,
	  lc($true) => $true,
	  1 => $true,
	  lc(substr($false, 0, 1)) => $false,
	  lc($false) => $false,
	  0 => $false,
	})->[lc($value)];
	Value->Error("The value should be one of $true or $false") unless defined $sanitized_value;
	return parser::PopUp->DropDown([ $true, $false ], $sanitized_value, %options);
}
```

Here we use a HASH to list the valid true/false values and look up the given value (once it is localize).  If it isn't one of the allowed options, we give an error.

I'm hoping `substr()` handles UTF-8 properly so that if there is a multi-character UTF value, you get the proper number of bytes.